### ISSUE TITLE
[BUG] Can't use my keyboard

### ISSUE THREAD

**Issue by noadubg** [tofodroid/mimi-mod#100]
I'm using a Yamaha Genos and I can chose it in the mod midi list but i can't play note with it.

I tested midi input and everything is working but not with the mod.

**Environment :**
 - OS: windows 11
 - Mod Version: latest
 - Minecraft Version: 1.20.1

**Additional context**
Can you add me on discord to see debug steps : @zurdioz

**Comment by Dekaven1** [comment]
me too i can't play with my keyboard, i've nord electro 6d, in windows 11 OS in 1.20.1 in latest version

**Comment by Dekaven1** [comment]
> me too i can't play with my keyboard, i've nord electro 6d, in windows 11 OS in 1.20.1 in latest version

OK i'm just dumb i've forgot to set device in the device config !

**Comment by noadubg** [comment]
> > me too i can't play with my keyboard, i've nord electro 6d, in windows 11 OS in 1.20.1 in latest version
> 
> OK i'm just dumb i've forgot to set device in the device config !

Does it work now ?

**Comment by tofodroid** [comment]
Hello! Apologies for the super late reply. Life got super crazy for awhile and I had pretty much no time for the mod. It sounds like you got things working but please let me know if you have any other issues!

I'm also definitely open to suggestions on ways to simplify or better explain the process of connecting a MIDI keyboard to the mod!

**Comment by noadubg** [comment]
Hello, It’s fine I got things to do too. It’s not working for now but you can contact me on @zurdioz with discord so I can show you. Thx

**Comment by tofodroid** [comment]
Sounds good, I should have some time later today or tomorrow to reach out!

In the meantime I just added a new Wiki page for using MIDI Input Devices in version 4.0.0 so feel free to check it out and see if it provides any help:

https://github.com/tofodroid/mimi-mod/wiki/Using-a-MIDI-Device

**Comment by noadubg** [comment]
Hello I don't understand why it doesn't work I did everything good. Can you still contact me on discord, what is yours?

**Comment by tofodroid** [comment]
Hello again!

I have just uploaded version 4.0.5 to [CurseForge](https://www.curseforge.com/minecraft/mc-mods/mimi-mod) and [Modrinth](https://modrinth.com/mod/mimi/versions) that should hopefully resolve this issue! I found what I think should be a much more reliable way for MIMI to determine the correct audio output device, so fingers crossed this resolves this issue for everyone!

I will also be uploading a 1.20.1-3.1.4 version to CurseForge which will contain this fix and be functionally closer to the version used on Craft2Exile and some other modpacks.

The CurseForge uploads are still processing (can take up to 24hours 😞) but the Modrinth versions are available now!

Let me know if you still have any issues!
